file_lib/src/debug.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crossterm::event;
+use std::env;
+
+
+
+pub const DEBUG_ENV_VAR: &str = "DEBUG_DSC";
+
+/// If the `DEBUG_DSC` environment variable is set, and it contains the given command,
+/// then prompt the user to attach a debugger.
+/// 
+/// The environment variable is a comma-separated list of case-insensitive commands to enable for
+/// prompting to attach the debugger.
+/// 
+/// # Arguments
+/// 
+/// * `command` - The command to check for in the `DEBUG_DSC` environment variable.
+/// 
+pub fn check_debug(command: &String) {
+    if env::var("DEBUG_DSC").is_ok() {
+        let debug_args: Vec<String> = env::var(DEBUG_ENV_VAR)
+            .unwrap()
+            .split(',')
+            .map(|s| s.to_lowercase())
+            .collect();
+
+        if debug_args.contains(command) {
+            eprintln!(
+                "attach debugger to pid {} and press any key to continue",
+                std::process::id()
+            );
+            loop {
+                let event = event::read().unwrap();
+                if let event::Event::Key(_key) = event {
+                    break;
+                }
+                eprintln!("Unexpected event: {event:?}");
+            }
+        }
+    }
+}
```

Comment:
We should probably pull this into it's own crate to be used by different projects within this repo, but that can be done as a separate PR

Reply:
Yeah, I wasn't sure where to put it. I didn't know if it belonged in `dsc_lib` or a standalone create, or something new like `dsc_utils`, and figured we could move it later. I have a similar refactor to break out the common code for reading STDIN into a function.

Reply:
@miroman9364 can you open an issue so we refactor that common code, probably a `dsc_utils` crate makes sense.